open-hackathon-server/src/hackathon/hk8s/k8s_service_adapter.py

```diff
@@ -44,12 +44,20 @@ def __init__(self):
         return
 
     def create_k8s_deployment_with_yaml(self, yaml, name, namespace):
-        k8s_api = utils.create_from_yaml(self.k8s_client, yaml)
-        resp = k8s_api.read_namespaced_deployment(name, namespace)
-        return str(resp.status)
+        try:
+            k8s_api = utils.create_from_yaml(self.k8s_client, yaml)
+            #see https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/AppsV1Api.md#read_namespaced_deployment
+            resp = k8s_api.read_namespaced_deployment(name, namespace)
+            return format(deps.metadata.name)
+        except Exception as e:
+            if 'Exists' in e:
```

Comment:
Using string comparing doesn't sound robust. Do they have a http status code?

Reply:
agree, but i don't know how to interpret 'e' and get status code, it's 409. however, 409 is not robust either, what if kubernetes changes?